cmd/manager/main.go

```diff
@@ -56,7 +58,14 @@ func main() {
 		log.Sugar().Fatalf("Failed to configure logger: %v", err)
 	}
 
-	if !hasRequiredVariables(log, construct.AgentImageEnv, construct.VersionUpgradeHookImageEnv, construct.ReadinessProbeImageEnv) {
+	if !hasRequiredVariables(
+		log,
+		construct.MongodbRepoUrlEnv,
+		construct.MongodbImageEnv,
```

Comment:
This is probably the most significant change in this PR: rest is moving things around and updating tests where applicable.

These env vars were not required before, but from the code it seems like they actually are required. So I'm adding them here. Would be good to get a sanity check here. Am I missing something? Is it actually possible and reasonable to run without them?